homeassistant/components/tplink_omada/switch.py

```diff
@@ -110,3 +175,78 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         self.port_details = self.coordinator.data[self.port_id]
         self._refresh_state()
+
+
+@dataclass
+class GatewayPortSwitchConfig:
+    """Config for a toggle switch derived from a gateway."""
+
+    port_number: int
```

Comment:
Move the port number out and pass it directly to the entity instead.

Reply:
Done